### PR TITLE
Added new `skip_waiting_for_build_processing` option

### PR DESCRIPTION
**Pull request by KrauseFx**
This allows the user to skip the processing time and/or the submission separately

Fixes https://github.com/fastlane/fastlane/issues/1463

### CODE REVIEW

File: pilot/lib/pilot/build_manager.rb [hunk 1]
```diff
@@ -18,18 +18,27 @@ def upload(options)
...
+        UI.important("Skip waiting for build processing")
+        UI.important("This means, no changelog will be set and no build will be distributed to testers")
+        return
       end
+
+      UI.message("If you want to skip waiting for the processing to be finished, use the `skip_waiting_for_build_processing` option")
+      uploaded_build = wait_for_processing_build # this might take a while
+      # First, set the changelog (if necessary)
+      uploaded_build.update_build_information!(whats_new: options[:changelog])
+      UI.success "Successfully set the changelog for build"
```
Comment (mpirri) [inline pilot/lib/pilot/build_manager.rb]: Do we want to do this/show this message if the user hasn't provided a changelog/what-to-test?

File: pilot/lib/pilot/build_manager.rb [hunk 2]
```diff
@@ -18,18 +18,27 @@ def upload(options)
...
-        UI.user_error!("Error uploading ipa file, more information see above")
+      if config[:skip_waiting_for_build_processing]
+        UI.important("Skip waiting for build processing")
+        UI.important("This means, no changelog will be set and no build will be distributed to testers")
+        return
       end
+
+      UI.message("If you want to skip waiting for the processing to be finished, use the `skip_waiting_for_build_processing` option")
+      uploaded_build = wait_for_processing_build # this might take a while
+      # First, set the changelog (if necessary)
```
Comment (mpirri) [inline pilot/lib/pilot/build_manager.rb]: This field is called "What to Test" on iTunesConnect and I've seen some confusion among users about what "What's New" or "Changelog" maps to in the iTC interface. Does it make sense to change these options (or add new options/deprecate the old ones) with a more consistent name?

File: pilot/lib/pilot/build_manager.rb [hunk 3]
```diff
@@ -18,18 +18,27 @@ def upload(options)
...
-          distribute_build(uploaded_build, options)
+      UI.message("Successfully uploaded the new binary to iTunes Connect")
 
-          UI.message("Successfully distributed build to beta testers 🚀")
-        end
-      else
-        UI.user_error!("Error uploading ipa file, more information see above")
+      if config[:skip_waiting_for_build_processing]
+        UI.important("Skip waiting for build processing")
+        UI.important("This means, no changelog will be set and no build will be distributed to testers")
```
Comment (mpirri) [inline pilot/lib/pilot/build_manager.rb]: From a grammatical standpoint, I think the comma after "This means" is unnecessary.  You might want to change it to a "that" - i.e., "This means that no changelog..."

File: pilot/lib/pilot/build_manager.rb [hunk 4]
```diff
@@ -18,18 +18,27 @@ def upload(options)
       transporter = FastlaneCore::ItunesTransporter.new(options[:username])
       result = transporter.upload(app.apple_id, package_path)
 
-      if result
-        UI.message("Successfully uploaded the new binary to iTunes Connect")
+      unless result
+        UI.user_error!("Error uploading ipa file, more information see above")
```
Comment (mpirri) [inline pilot/lib/pilot/build_manager.rb]: "more information" -> "for more information"

**Comment by mpirri** [comment]
Few minor things, but this looks awesome! It will be great to finally address this problem!

File: pilot/lib/pilot/build_manager.rb (hunk 1 above)
Comment (KrauseFx) [inline pilot/lib/pilot/build_manager.rb]: Very good point

**Comment by KrauseFx** [comment]
Thanks for your feedback @mpirri, I just added https://github.com/fastlane/fastlane/commit/6b30821b30abd0da6a6600e54ab8d5a151d5609d

**Comment by KrauseFx** [comment]
And the version bump https://github.com/fastlane/fastlane/pull/4811

**Comment by mpirri** [comment]
Looks great, ship it!

**Pull request merged**